specs/Deployment/MSIXPackageVersioning.md

```diff
@@ -0,0 +1,517 @@
+version of the project.
+
+***Recommendation:*** Option A and C (when necessary).
+
+All tags have a long name (Option A) and a short (1-2 characters) name (Option C).
+
+Option C is used when Option A doesn't work due to package Name length restrictions.
+Currently WARddlm is the only instance using Option C.
+
+## 2.4. Decision 4: Version Encoding
+
+What data's encoded in the MSIX Version, and how?
+
+The following notation is used to describe the possible options
+
+* X = Major version
+* N = Minor version
+* P = Patch (per SemVer definition i.e. the 3rd field in a version)
+* Y = Year
+* M = Month
+* D = Day
+* E = Elapsed days since an epoch e.g. if Epoch=2021/01/01 then 2021/6/4=154
+* B = Build number (reset to zero when Major or Minor changes)
+* 0 = Zero
+
+Single character indicates variable-length without leading zeros.
+Multiple characters indicates fixed-length with leading zeroes if necessary. For example, Y=1 vs YY=01.
+
+NOTE: MSIX version's 4th field (Revision) is reserved for security updates.
+
+Examples are provided for the following theoretical versions...
+* Windows App SDK 0.8.1, date=2021/6/4, build=123, elapsed=154
+* Windows App SDK 1.2.3, date=2021/12/31, build=123, elapsed=364
+* Windows App SDK 17.14.3, date=2031/10/20, build=123, elapsed=3944
+
+|Option|Syntax|Example|Information|Comments|
+| --- | :--- | :---: | :---: | :---: |
+|A|X.N.P.0|0.8.1.0<br>1.2.3.0<br>17.14.3.0|Major Minor Patch||
+|B|X.NYYMM.DDPPP.0|0.82106.04001.0<br>1.22112.31003<br>--error--|Major Minor Patch Date|Minor max is 6<br>Patch max is 999<br>WinUI 2.x used this encoding<br> Windows App SDK 0.5 used this encoding|
+|C|XNN.PYYMM.DDBBB.0|8.12106.04123.0<br>102.32112.31123.0<br>1714.33110.20123.0|Major Minor Patch Date Build|Minor max is 99<br>Patch max is 6<br>Build max is 999|
+|D|NPPP.E.B.0|8001.154.123<br>2003.364.123<br>17014.3944.123.0|Minor Patch Elapsed Build|Minor max is 64<br>Patch max is 999<br>Build max is 65535|
+|E|NPP.E.B.0|801.154.123<br>203.364.123<br>1714.3944.123.0|Minor Patch Elapsed Build|Minor max is 99<br>Patch max is 99<br>Build max is 65535|
+
+***Recommendation:*** Option D with Epoch=January 1, 2021.
+
+Option A works for a simple release strategy but doesn't work when there's regular
+(e.g. daily) builds.
+
+Option B worked for WinUI 2.x and Windows App SDK 0.5 but is too limiting given it can't support
+Minor versions beyond 6. It's also a complicated encoding scheme with date injected between Minor
+and Patch (not readily legible).
+
+Option C is too limiting given it can't support Patch versions beyond 6.
+
+Option D relies on the package Name including Major version. This leaves sufficient space in the
+version to encode up to 99 Minor and 99 Patch versions. Date encoding as an Epoch reduces the size
+while supporting 20+ years with only 4 digits.
+
+Option E is a variant of Option D except Patch version is restricted to 2 digits, but it's an
+unnecessary simplification.
+
+### 2.4.1. Windows App SDK 0.5 Notation
```

Comment:
Is this here for historical reasons? We're past 0.8 at this point.

Reply:
Yes, historical. That's why it's in a sub-section and not part of any option for consideration. We started with that, quickly saw the limitations and gamed out alternatives to land on our current solution. But knowing this history helps many to understand "Why is it so complicated? Why can't it be simple?".